nucleus/deployment/admin/src/main/java/org/glassfish/deployment/admin/AddLibraryCommand.java

```diff
@@ -117,17 +135,31 @@ public void execute(AdminCommandContext context) {
             List<UnprocessedChangeEvent> unprocessed = 
                 new ArrayList<UnprocessedChangeEvent>();
 
-            StringBuffer msg = new StringBuffer();
+            StringBuilder msg = new StringBuilder();
+            
+            ClassLoader commonLoader = commonClsLdr.getCommonClassLoader();
+            CurrentBeforeParentClassLoader loader = null;
+            if (commonLoader instanceof CurrentBeforeParentClassLoader){
+                loader = (CurrentBeforeParentClassLoader) commonLoader;
+            }
 
             for (File libraryFile : files) {
                 if (libraryFile.exists()) {
-                    DeploymentCommandUtils.renameUploadedFileOrCopyInPlaceFile(
+                    logger.log(Level.SEVERE, "ready to add new library");
+                    File result = DeploymentCommandUtils.renameUploadedFileOrCopyInPlaceFile(
                         libDir, libraryFile, logger, env);
-                    PropertyChangeEvent pe = new PropertyChangeEvent(libDir, 
-                        "add-library", null, libraryFile);
-                    UnprocessedChangeEvent uce = new UnprocessedChangeEvent(
-                        pe, "add-library");
-                    unprocessed.add(uce);
+                    
+                    if (loader != null ){
```

Comment:
Don't you have to choose a classloader based on the type i.e. ext, common or app?